File: Data_Structure/DS_Using_Python/userDefine_Doubly_LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.prev = None
		self.next = None

class LinkedList:
	def __init__(self):
		self.head = None
		
	def traverseData(self):
		temp = self.head
		while temp is not None:
			print(temp.data, end = ' ')
			temp = temp.next
			
	def insertFirst(self, data):
					ptr = Node(data)
					if self.head is None:
						ptr.data = data
						ptr.prev = None
						ptr.next = None
						self.head = ptr
					else:
						ptr.data = data
						ptr.prev = None
						ptr.next = self.head
						self.head = ptr
			
	def userDefineData(self):
		print("\nEnter the number of nodes:")
		noOfNode = int(input())
		print("\nEnter the item:")
		for i in range(noOfNode):
		  	 data = int(input())
		  	 self.insertFirst(data)
		
llist = LinkedList()

llist.userDefineData()
print("\nLinked List is")
llist.traverseData()

File: Data_Structure/LinkedList/userDefine_Doubly_LinkedList.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;
    struct node * prev;
};

struct node *head = NULL;

void traverseData()
{
    struct node* ptr;
    ptr = head;
    while(ptr != NULL)
    {
        printf("%d ", ptr -> data);
        ptr = ptr -> next;
    }
}

//New node will insert before the next node
void insertData(int data)
{
    struct node* ptr;
    ptr = (struct node*)malloc(sizeof(struct node));
    if(head == NULL)
    {
         ptr -> data = data;
         ptr -> next = NULL;
         ptr -> prev = NULL;
         head = ptr;
    }
    else
    {
         ptr -> data = data;
         ptr -> prev = NULL;
         ptr -> next = head;
         head -> prev = ptr;
         head = ptr;
     }
}

void userDefinedData()
{
    int data, noOfNode;
    printf("\nEnter the number of nodes:\n");
    scanf("%d",&noOfNode);
    printf("\nEnter the item:\n");
    for(int i=0; i<noOfNode; i++)
    {
        scanf("%d",&data);
        insertData(data);
    }
}

int main()
{
    userDefinedData();
    printf("Linked List is\n");
    traverseData();
    
    return 0;
}